Upgrade virt-top version to 1.1.1.

durgajagadeesh opened the pull request:
<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
Upgrade virt-top version to 1.1.1.

###### Change Log  <!-- REQUIRED -->
<!-- Detail the changes made here. -->
<!-- Please list any packages which will be affected by this change, if applicable. -->
<!-- Please list any CVES fixed by this change, if applicable. -->
- Bump up the version.
- License verified.

###### Does this affect the toolchain?  <!-- REQUIRED -->
<!-- Any packages which are included in the toolchain should be carefully considered. Make sure the toolchain builds with these changes if so. -->
<!-- Update: manifests/package/toolchain_*.txt, pkggen_core_*.txt, update_manifests.sh -->
<!-- To validate: make clean; make workplan REBUILD_TOOLCHAIN=y DISABLE_UPSTREAM_REPOS=y CONFIG_FILE="" ... -->
**NO**

###### Associated issues  <!-- optional -->
<!-- Link to Github issues if possible. -->
<!-- you can use "fixes #xxxx" to auto close an associated issue once the PR is merged -->
- #xxxx

###### Links to CVEs  <!-- optional -->
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->
- Pipeline build id: xxxx

durgajagadeesh commented:
Hi,
This package is ready for review, It was built successfully in my VM.
<img width="254" alt="image" src="https://github.com/user-attachments/assets/9840cd2b-9e91-4f00-83e6-a7bc480274c3" />

Thank you!

kgodara912 reviewed (commented):
Good for next stage: buddy build pipeline.

cyberbandya007 reviewed (approved):
Buddy Build Successful.
Changes LGTM.

cyberbandya007 commented on SPECS-EXTENDED/virt-top/virt-top-1.0.9-processcsv-documentation.patch:
```diff
@@ -0,0 +1,26 @@
+ 
+  virt-top --csv >(split -d -l 1000 - output.csv.)
+ 
++RHEL provides a short Python script called C<processcsv.py> which
```
@durgajagadeesh , Can you remove this documentation patch?

durgajagadeesh replied:
Hi @cyberbandya007,
I have updated the patch according to your suggestions from our offline discussion.
-Thank you!.

cyberbandya007 reviewed (approved):
LGTM

kgodara912 reviewed (changes requested):
Please check two things, using autosetup and putting patch file in the macro instead of having patch command individually. Secondly, please check other spec files for `gpgverify`, as we are already verifying the source signature what purpose is solved that command.

durgajagadeesh commented:
> Please check two things, using autosetup and putting patch file in the macro instead of having patch command individually. Secondly, please check other spec files for `gpgverify`, as we are already verifying the source signature what purpose is solved that command.

Hi @kgodara912,
Got it. Working on the changes for autosetup/macro and investigating gpgverify now. Will update the PR shortly.
-Thank you!

durgajagadeesh commented:
> Please check two things, using autosetup and putting patch file in the macro instead of having patch command individually. Secondly, please check other spec files for `gpgverify`, as we are already verifying the source signature what purpose is solved that command.

Hi @kgodara912,
I have implemented the changes based on your suggestions.
After running autosetup, the build completed successfully on my local VM.

I have removed the conditional patches (P1 and P5) that were not used by Azure Linux virt-top.

Additionally, I captured the log files before and after running autosetup:
<img width="622" height="372" alt="image" src="https://github.com/user-attachments/assets/8bf8a163-f049-437a-8cd0-1c90e4b358b5" />
i) The first screenshot shows the state before applying autosetup.
<img width="1377" height="467" alt="image" src="https://github.com/user-attachments/assets/562b5878-ab4c-49db-8c27-270a825c09da" />
ii) The second screenshot shows the state after applying autosetup.
<img width="1463" height="365" alt="image" src="https://github.com/user-attachments/assets/ab9853af-f47f-4010-b0b2-6ad04171e67e" />


Please note that I have only removed RHEL-related(P1) and unused patch(P5) files only.
Please let me know if I need to make any further changes.
-Thank you!

kgodara912 commented:
[Buddy build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=941721&view=results) will the latest changes.

durgajagadeesh commented:
> [Buddy build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=941721&view=results) will the latest changes.

Hi @kgodara912,
The buddy build completed successfully.  
<img width="625" height="556" alt="image" src="https://github.com/user-attachments/assets/bc8d26c0-0e9b-47b9-a9ed-abe096b2c67f" />
-Thank you!

durgajagadeesh commented:
> Please remove the non-required files like keyring.

Hi @kgodara912,
I have made changes based on your suggestions. The build is completed successfully and patches have been applied cleanly in my local VM.
<img width="1475" height="252" alt="image" src="https://github.com/user-attachments/assets/5d9fd0f1-f075-4e80-92a7-c63807aa354d" />
Please let me know if I need to make any further changes.
-Thank you!

durgajagadeesh commented:
> BB - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=942483&view=results

Hi @cyberbandya007,
The buddy build completed successfully,
<img width="701" height="708" alt="image" src="https://github.com/user-attachments/assets/8018a74e-18e6-4d02-ad5e-e77702e3715a" />
And the patches applied cleanly.
<img width="1662" height="297" alt="image" src="https://github.com/user-attachments/assets/f938dfed-8407-470e-a07c-fab998ad89f7" />
<img width="1892" height="70" alt="image" src="https://github.com/user-attachments/assets/e459a0e8-4b1e-4c35-8096-f3c2e5f5be19" />

-Thank you!

kgodara912 commented on SPECS-EXTENDED/virt-top/virt-top.spec:
```diff
@@ -1,49 +1,40 @@
 BuildRequires:  perl(Pod::Perldoc)
 BuildRequires:  gawk
-
+BuildRequires:  gnupg2
```
Check if this is still needed.

durgajagadeesh replied:
Hi @kgodara912,
I have started working on this and will follow up with a potential solution soon.
Thank you.

durgajagadeesh replied:
Hi @kgodara912,
I've incorporated the changes as per your above suggestion.
Please let me know if you require any additional modifications.
Thank you.

kgodara912 reviewed (changes requested):
The `virt-top` doesn't work yet.
```
$ virt-top
libvirt: VIR_ERR_SYSTEM_ERROR: VIR_FROM_RPC: binary '/usr/sbin/libvirtd' does not exist in $PATH: No such file or directory
NB: If you want to monitor a local hypervisor, you usually need to be root

Also,
$ sudo virt-top
libvirt: VIR_ERR_SYSTEM_ERROR: VIR_FROM_RPC: Failed to connect socket to '/var/run/libvirt/libvirt-sock-ro': No such file or directory
```
It should have a `Requires` for `libvirt` in spec file which will install and create necessary dependencies for `virt-top` to run. After installation of `libvirt`, `virt-top` is able to give some output. Please verify from your end and make necessary changes.

cyberbandya007 commented:
> The `virt-top` doesn't work yet.
> 
> ```
> $ virt-top
> libvirt: VIR_ERR_SYSTEM_ERROR: VIR_FROM_RPC: binary '/usr/sbin/libvirtd' does not exist in $PATH: No such file or directory
> NB: If you want to monitor a local hypervisor, you usually need to be root
> 
> Also,
> $ sudo virt-top
> libvirt: VIR_ERR_SYSTEM_ERROR: VIR_FROM_RPC: Failed to connect socket to '/var/run/libvirt/libvirt-sock-ro': No such file or directory
> ```
> 
> It should have a `Requires` for `libvirt` in spec file which will install and create necessary dependencies for `virt-top` to run. After installation of `libvirt`, `virt-top` is able to give some output. Please verify from your end and make necessary changes.

As I have checked fedora, we get similar error and I don't think fedora also installs full libvirt. It installs libvirt-libs.

[root@6d2902a14fff /]# yum install virt-top
Updating and loading repositories:
 Fedora 42 openh264 (From Cisco) - x86_64                                                                                     100% |   2.5 KiB/s |   5.8 KiB |  00m02s
 Fedora 42 - x86_64 - Updates                                                                                                 100% |   4.4 MiB/s |  19.8 MiB |  00m05s
 Fedora 42 - x86_64                                                                                                           100% |   5.1 MiB/s |  35.4 MiB |  00m07s
Repositories loaded.
Package                                                 Arch          Version                                                 Repository                          Size
Installing:
 virt-top                                               x86_64        1.1.2-1.fc42                                            fedora                           3.6 MiB
Installing dependencies:
 cyrus-sasl-gssapi                                      x86_64        2.1.28-30.fc42                                          fedora                          40.2 KiB
 libnl3                                                 x86_64        3.11.0-3.fc42                                           fedora                           1.0 MiB
 libssh2                                                x86_64        1.11.1-3.fc42                                           fedora                         322.7 KiB
 libvirt-libs                                           x86_64        11.0.0-4.fc42                                           updates                         27.2 MiB
 libwsman1                                              x86_64        2.7.2-12.fc42                                           fedora                         355.0 KiB
 numactl-libs                                           x86_64        2.0.19-2.fc42                                           fedora                          52.9 KiB

Transaction Summary:
 Installing:         7 packages

Total size of inbound packages is 7 MiB. Need to download 7 MiB.
After this operation, 33 MiB extra will be used (install 33 MiB, remove 0 B).
Is this ok [y/N]: y
[1/7] libnl3-0:3.11.0-3.fc42.x86_64                                                                                           100% | 351.6 KiB/s | 357.2 KiB |  00m01s
[2/7] libssh2-0:1.11.1-3.fc42.x86_64                                                                                          100% | 743.5 KiB/s | 147.2 KiB |  00m00s
[3/7] virt-top-0:1.1.2-1.fc42.x86_64                                                                                          100% | 813.0 KiB/s |   1.1 MiB |  00m01s
[4/7] libvirt-libs-0:11.0.0-4.fc42.x86_64                                                                                     100% |   3.6 MiB/s |   5.1 MiB |  00m01s
[5/7] libwsman1-0:2.7.2-12.fc42.x86_64                                                                                        100% | 633.5 KiB/s | 129.2 KiB |  00m00s
[6/7] numactl-libs-0:2.0.19-2.fc42.x86_64                                                                                     100% | 162.9 KiB/s |  31.3 KiB |  00m00s
[7/7] cyrus-sasl-gssapi-0:2.1.28-30.fc42.x86_64                                                                               100% |  57.0 KiB/s |  26.5 KiB |  00m00s
----------------------------------------------------------------------------------------------------------------------------------------------------------------------
[7/7] Total                                                                                                                   100% |   3.6 MiB/s |   6.8 MiB |  00m02s
Running transaction
[1/9] Verify package files                                                                                                    100% | 170.0   B/s |   7.0   B |  00m00s
[2/9] Prepare transaction                                                                                                     100% | 437.0   B/s |   7.0   B |  00m00s
[3/9] Installing cyrus-sasl-gssapi-0:2.1.28-30.fc42.x86_64                                                                    100% |   5.7 MiB/s |  41.1 KiB |  00m00s
[4/9] Installing numactl-libs-0:2.0.19-2.fc42.x86_64                                                                          100% |   4.8 MiB/s |  53.8 KiB |  00m00s
[5/9] Installing libwsman1-0:2.7.2-12.fc42.x86_64                                                                             100% |  49.9 MiB/s | 357.9 KiB |  00m00s
[6/9] Installing libssh2-0:1.11.1-3.fc42.x86_64                                                                               100% |  52.8 MiB/s | 324.4 KiB |  00m00s
[7/9] Installing libnl3-0:3.11.0-3.fc42.x86_64                                                                                100% |  73.1 MiB/s |   1.0 MiB |  00m00s
[8/9] Installing libvirt-libs-0:11.0.0-4.fc42.x86_64                                                                          100% | 426.0 MiB/s |  27.3 MiB |  00m00s
[9/9] Installing virt-top-0:1.1.2-1.fc42.x86_64                                                                               100% |  27.4 MiB/s |   3.6 MiB |  00m00s
Complete!
[root@6d2902a14fff /]# virt-top
libvirt: VIR_ERR_OPERATION_UNSUPPORTED: VIR_FROM_REMOTE: Operation not supported: Cannot use direct socket mode if no URI is set. For more information see https://libvirt.org/kbase/failed_connection_after_install.html
[root@6d2902a14fff /]# sudo virt-top
libvirt: VIR_ERR_OPERATION_UNSUPPORTED: VIR_FROM_REMOTE: Operation not supported: Cannot use direct socket mode if no URI is set. For more information see https://libvirt.org/kbase/failed_connection_after_install.html

The error you are seeing is expected since the virtualization ecosystem is not configured for virt-top to work. However if you insist, we can add hard dependency on libvirt.

kgodara912 reviewed (approved):
Buddy build is successful. Generated rpm package installs fine and `virt-top` command runs fine. LGTM.
